lib/gli/config_loader.rb

```diff
@@ -0,0 +1,25 @@
+require 'yaml'
+
+module GLI
+  class ConfigLoader
+    def self.load(path)
+      File.open(path) { |file| apply_indifferent_access_to(YAML::load(file)) }
+    end
+
+    # adapted from https://gist.github.com/Integralist/9503099
+    def self.apply_indifferent_access_to(obj)
+      return obj.reduce({}) do |memo, (k, v)|
```

Comment:
Sorry to nitpick, but can you use a more traditional `if` construct?  Trailing `if`s for multiple lines is really confusing to me.